src/core/models.py

```diff
@@ -380,10 +386,18 @@ def host(self) -> str:
 
         return host
 
+    @property
+    def host(self) -> str:
+        """The hostname for the unit."""
+        if self.substrate == "vm":
+            return self.internal_address
+        else:
+            return self.node_ip or self.internal_address
```

Comment:
**question:** It's been a while since I looked that hard at ZK, but where else are we using `host`?

I ask, because IIRC we use actual 'hostnames' for the internal ZK DNS by patching `/etc/hosts`. I want to make sure we are:

1. Only using `hostname IPaddress` in 'internal' DNS. i.e not passing node/lb address 
2. Only using node/lb address in 'client' relations

Reply:
Actually, I think we are better off removing this altogether because it's confusing and hardly used in the first place. In VM, we use it sparingly in `upgrade.py` so I'll just replace it altogether by `internal_address`

Reply:
Removed in 53279c0aa2bb6dee4c3426e29d3bbc8e808aa9eb